### ISSUE TITLE
Kernel dies when I try to run this code.

### ISSUE THREAD

**Issue by bayismet** [Whadup/weightedFastText#1]
Hello there!

I suppose I managed all the data, labels and weights. The last code starts to work but after a while (like 2mins) kernel dies. I checked "htop", I got 19.6G total RAM and only 5 G'ish is used so I don't think it's based on RAM. I use Ubuntu 20.04LTS and just updated all the packages. 

Apart from that, while I'm managing my data, if I open "htop", it gives me this Ubuntu error:
![The error I take if I open "htop" while I'm parsing my data](https://user-images.githubusercontent.com/28671373/119261915-125cb280-bbe2-11eb-962d-ebf0261838fb.png)

Here are some images of my data, labels and the code:

![My data (a list), labels (a numpy array) and weights (a numpy array)](https://user-images.githubusercontent.com/28671373/119262058-8eef9100-bbe2-11eb-984b-99abe491643a.png)

![The code I execute](https://user-images.githubusercontent.com/28671373/119262095-ab8bc900-bbe2-11eb-8180-a891c27c2ecd.png)

![And kernel dies](https://user-images.githubusercontent.com/28671373/119262118-c3634d00-bbe2-11eb-9d68-1810dc38d1ee.png)

![Terminal, after kernel dies](https://user-images.githubusercontent.com/28671373/119261960-391ae900-bbe2-11eb-89a3-96f920477620.png)

Thanks!

**Comment by Whadup** [comment]
can you check on a tiny fraction of the data?

**Comment by bayismet** [comment]
I used 1190 samples (approx. 0,95% of my dataset) and I got the following result, though I'm not sure what should I understand from this result.
![Screenshot from 2021-05-24 00-45-43](https://user-images.githubusercontent.com/28671373/119277673-b3229080-bc29-11eb-9a7b-1d9d687ecd8a.png)

**Comment by Whadup** [comment]
Seems like your labels are float rather than into values.On May 23, 2021 23:48, İsmet Bayramoğlu ***@***.***> wrote:
I used 1190 samples (approx. 0,06% of my dataset) and I got the following result, though I'm not sure what should I understand from this result.

**Comment by bayismet** [comment]
Okay then, I "astype(int)"ed my labels. But I still got this output:

![Screenshot from 2021-05-24 11-52-25](https://user-images.githubusercontent.com/28671373/119322774-a6805580-bc86-11eb-85e3-0ab1888b34d1.png)
BTW, I'm expecting a model that trained by the weights I passed, am I right?

**Comment by Whadup** [comment]
Well actually you're running a grid search for best parameters, so what you get as a result is not the model, but a GridSearchCV object ( https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.GridSearchCV.html )

If you just want the model, you can use `ft.fit(X, y, sample_weights=weights)`

**Comment by bayismet** [comment]
ft.ft(...) gave me the product that I want (hooray!). If I use 1190 (119 classes, 10 samples from each class) samples, I can fit it and get scores. But when I try to use my whole dataset (124817 samples), the kernel just dies.

**Comment by Whadup** [comment]
Well in my experience FastText isn't that hardware hungry.
Have you tried using python without the jupyter kernel?

**Comment by bayismet** [comment]
Yes I did, but the output is still same...
![python Terminal](https://user-images.githubusercontent.com/28671373/119494967-8fb32f00-bd6a-11eb-9953-e48d58c2ee1f.png)

**Comment by Whadup** [comment]
Alright...the "loading precomputed" message confuses me. Are you really trying to load precomputed word embeddings?

**Comment by bayismet** [comment]
Yes we have a pretrained vector file and I'm also using that (a .vec file). But I also tried without using that, still having the same issue.

Shouldn't I use that vector?

**Comment by Whadup** [comment]
I think when the tiny fraction of the data works with pretrained vectors, then the file is not the issue.

Could you paste how you create the FastTextEstimator object and call fit on it?
From the log files and checking facebooks issues, it might be that the training loop throws an exception caused by some numerical issues caused by weird hyperparameter settings. Like the 500 epochs in your previous screenshot, that is way too many, maybe try 5.

**Comment by bayismet** [comment]
`ft = FastTextEstimator(wordNgrams=2,
                       epoch=500,
                       dim=50,
                       verbose=1,
                       pretrainedVectors="/data/data/models/w2v/mecelle-201908-1122.vec",
                       n_jobs=12)
ft.fit(X, y, sample_weight=weights)`

We also use FastText supervised to train a model the same data. So I just copied the parameters from there. I'll also lower the epoch now and try again. I'll let you know if it works.

Edit: 5 - 10 epoch worked but my score is lower than normal FastText model. Did I do something still wrong?

![Screenshot from 2021-05-25 22-21-12](https://user-images.githubusercontent.com/28671373/119556468-b774b800-bda7-11eb-8bdc-a3b868b75349.png)

**Comment by Whadup** [comment]
Well the score can be lower with class weights, e.g. when you sacrifice performance on the frequent classes for performance increases in the infrequent classes. I suggest you compute some per-class metrics, e.g. https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html and figure out what's going on and if that is what you intended your weights to do.

That being said, that your train and test performance differs that much is cause for concern, apparently your model is overfitting. Maybe tune your hyperparameters further. Or the weights cause problems?

**Comment by bayismet** [comment]
I'll try it with different train&test set, and also with different parameters, but turns out code is working fine! So I believe this issue is closed. Thank you for your help!
